bash_103_combining_commands/bash_103_combining_commands.md

```diff
@@ -0,0 +1,463 @@
+
+```
+sort Animals.csv | tail -3 > last.txt
+```
+
+This command doesn't give any output, but take a look at the contents of your new file `last.txt` and confirm that it contains what you expected.
+
+The double arrow `>>` which appends output to a file can also be combined with pipes. Try this line:
+
+```
+tail -3 Animals.csv | sort >> last.txt
+```
+
+Now the file `last.txt` should contain six lines:
+
+```
+red panda,mammal
+tiger,mammal
+wolf,mammal
+blue morpho,insect
+tiger,mammal
+wolf,mammal
+```
+
+### `uniq` revisited
+
+Now that the file `last.txt` has some repeated lines in it, we can get useful information out of the command `uniq`. This command outputs the lines of a file, but only those lines that are distinct from the line directly above.
+
+```
+uniq last.txt
+```
+
+Since all lines in `last.txt` are different from the line immediately preceding them, this isn't particularly useful. However when combined with the `sort` function, we can eliminate repeated lines:
+
+```
+sort last.txt | uniq
+```
+
+Now there are no repeated lines! They are also in alphabetical order since we first sorted alphabetically.
+
+If we want to know how many copies of each line were in the original file, the option `-c` for "count" will output the lines with each preceded by the number of occurrences:
+
+```
+sort last.txt | uniq -c
+```
+### Chaining multiple pipes
+
+Piping the output of one command into another is extremely powerful, and chaining multiple pipes to get a **pipeline** only increases the number of things you can do with pipes!
+
+Maybe you want to know how many unique animals are in the file `last.txt`. We saw on the last page how to output the list of unique lines using `sort` followed by `uniq`. If we want to know how many (unique) lines there are, we can use the word count function `wc` with the option `-l` for number of lines:
+
+```
+sort last.txt | uniq | wc -l
+```
+
+This gives us the single numeric output `4`; there are 4 lines corresponding to the 4 unique animals listed in `last.txt`.
+
+What if we want to know which animal has the shortest scientific name? This also becomes a three step process:
+
+1. Count the characters in each `.txt` file with `wc -m *.txt`. For further explanation of how `*.txt` calls all of the `.txt.` files, see the [lesson on searching and organizing files](https://liascript.github.io/course/?https://raw.githubusercontent.com/arcus/education_modules/main/bash_command_line_102/bash_command_line_102.md).
+2. Sort the output by number with `sort -n`.
+3. Output the first line with `head -1`.
+
+Putting these steps together:
+
+```
+wc -m *.txt | sort -n | head -1
+      10 indian_cobra.txt
+```
+The Indian cobra, scientific name _Naja naja_ has the shortest character count of any of the animals listed here.
+
+<div class = "learnmore">
+**Why is the character count for `Naja naja` 10?**
```

Comment:
Formatting is funky here, the newline after the bolded text is not appearing.

Reply:
sooo many newlines!!!